astropy/coordinates/builtin_frames/intermediate_rotation_transforms.py

```diff
@@ -61,13 +62,101 @@ def cirs_to_itrs_mat(time):
     return erfa.c2tcio(np.eye(3), era, pmmat)
 
 
+def tete_to_itrs_mat(time):
+    # compute the polar motion p-matrix
+    xp, yp = get_polar_motion(time)
+    sp = erfa.sp00(*get_jd12(time, 'tt'))
+    pmmat = erfa.pom00(xp, yp, sp)
+
+    # now determine the greenwich apparent siderial time for the input obstime
+    # we use the 2006A model for consistency with RBPN matrix use in GCRS <-> TETE
+    ujd1, ujd2 = get_jd12(time, 'ut1')
+    jd1, jd2 = get_jd12(time, 'tt')
+    gast = erfa.gst06a(ujd1, ujd2, jd1, jd2)
+
+    # c2tcio expects a GCRS->CIRS matrix, but we just set that to an I-matrix
+    # because we're already in CIRS equivalent frame
+    return erfa.c2tcio(np.eye(3), gast, pmmat)
+
+
 def gcrs_precession_mat(equinox):
     gamb, phib, psib, epsa = erfa.pfw06(*get_jd12(equinox, 'tt'))
     return erfa.fw2m(gamb, phib, psib, epsa)
 
 
 # now the actual transforms
 
+@frame_transform_graph.transform(FunctionTransformWithFiniteDifference, GCRS, TETE)
+def gcrs_to_tete(gcrs_coo, tete_frame):
+    # first apply GCRS self transform to correct time and observatory position/velocity
+    gcrs_coo2 = gcrs_coo.transform_to(GCRS(obstime=tete_frame.obstime, obsgeoloc=tete_frame.obsgeoloc,
```

Comment:
Maybe break such that `obsgeoloc` and `obsgeovel` align? (sorry, total nitpick...)

Reply:
done